share/spack/setup-env.sh

```diff
@@ -273,8 +273,15 @@ fi;
 #
 # set module system roots
 #
-_spack_pathadd DK_NODE    "${_sp_dotkit_root%/}/$_sp_sys_type"
-_spack_pathadd MODULEPATH "${_sp_tcl_root%/}/$_sp_sys_type"
+IFS=':' read -ra tcl_roots <<< "$_sp_tcl_roots"
```

Comment:
`read -a` is a bashism that will break with ZSH (equivalent here: `read -A`, pretty annoying…).

Reply:
That's new to me, good catch! We should clean that up

Reply:
I've attempted a fix [here](https://github.com/scheibelp/spack/pull/2).

Reply:
@scheibelp couldn't this be changed for ZSH users?